packages/ng-query/src/lib/tests/mutation.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { TestBed } from '@angular/core/testing';
+import { subscribeSpyTo } from '@hirez_io/observer-spy';
+import { UseMutation, MutationProvider } from '../mutation';
+import { errorMutator, flushPromises, successMutator } from './test-utils';
+
+describe('useMutation', () => {
+  let useMutation: UseMutation;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    useMutation = TestBed.inject(MutationProvider);
+  });
+
+  it('should be defined', async () => {
+    expect(useMutation).toBeTruthy();
+  });
+
+  it('should be in idle state initially', () => {
+    const mutation = useMutation((params) => successMutator(params));
+
+    expect(mutation.getCurrentResult()).toMatchObject({
+      isIdle: true,
+      isLoading: false,
+      isError: false,
+      isSuccess: false,
+    });
+  });
+
+  it('should change state after invoking mutate', async () => {
+    const mock = 'Mock data';
+    const mutation = useMutation((params: string) => successMutator(params));
+
+    mutation.mutate(mock);
+
+    const observerSpy = subscribeSpyTo(mutation.result$);
+    await flushPromises();
+    const result = observerSpy.getLastValue();
+
+    expect(result).toMatchObject({
+      isIdle: false,
+      isLoading: true,
+      isError: false,
+      isSuccess: false,
+      data: undefined,
+      error: null,
+    });
+  });
+
+  it('should return error when request fails', async () => {
+    const mutation = useMutation(errorMutator);
```

Comment:
if `mutationFn` throws error, this error is not caught and gets to zonejs. Tested in playground, is same there as well. @NetanelBasal

Reply:
>  this error is not caught and gets to zonejs

Can you elaborate more on this? What do you mean?

Reply:
my bad. Actually, the expected behavior is correct. Never mind. The only problem I'm left with this test case is that I cannot get rid of this log message even though the test case is successful. 

![image](https://user-images.githubusercontent.com/61333365/197865394-195428b6-21e1-43c8-bbc2-42b3affe5081.png)

Reply:
https://github.com/facebook/jest/issues/5785

Reply:
thank you, that worked pretty well